cmd/monitor/monitor.go

```diff
@@ -568,7 +569,15 @@ func renderMonitorUI(ctx context.Context, ec *ethclient.Client, ms *monitorStatu
 		ms.BlocksLock.RUnlock()
 		renderedBlocks = renderedBlocksTemp
 
-		skeleton.Current.Text = ui.GetCurrentText(skeleton.Current, ms.HeadBlock, ms.GasPrice, ms.PeerCount, ms.ChainID, rpcUrl)
+		// First initialization will render no gas price because the GasPriceChart will have no data.
+		if skeleton.GasPriceChart.Data == nil {
+			skeleton.Current.Text = ui.GetCurrentText(skeleton.Current, ms.HeadBlock, "--", ms.PeerCount, ms.ChainID, rpcUrl)
+		} else {
+			// Under normal cases, the gas price will be derived from the last element of the GasPriceChart with 2 decimal places precision.
+			gasPriceStr := strconv.FormatFloat(skeleton.GasPriceChart.Data[len(skeleton.GasPriceChart.Data)-1]/1000000000, 'f', 2, 64)
```

Comment:
- I would reuse the value from `metrics.GetMeanGasPricePerBlock(renderedBlocks)` instead of deriving it from the UI element.
- I'm not sure if we want to take the last element of the array (the last block shown). It would maybe make more sense to use the first element instead (the first block shown), don't you think?

```diff
diff --git a/cmd/monitor/monitor.go b/cmd/monitor/monitor.go
index 53beab5..e6cceec 100644
--- a/cmd/monitor/monitor.go
+++ b/cmd/monitor/monitor.go
@@ -569,12 +569,19 @@ func renderMonitorUI(ctx context.Context, ec *ethclient.Client, ms *monitorStatu
                ms.BlocksLock.RUnlock()
                renderedBlocks = renderedBlocksTemp
 
+               // Get mean gas price per block.
+               meanGasPricePerBlock := metrics.GetMeanGasPricePerBlock(renderedBlocks)
+               var meanGasPriceFirstBlock float64
+               if len(meanGasPricePerBlock) > 0 {
+                       meanGasPriceFirstBlock = meanGasPricePerBlock[0]
+               }
+
                // First initialization will render no gas price because the GasPriceChart will have no data.
                if skeleton.GasPriceChart.Data == nil {
                        skeleton.Current.Text = ui.GetCurrentText(skeleton.Current, ms.HeadBlock, "--", ms.PeerCount, ms.ChainID, rpcUrl)
                } else {
                        // Under normal cases, the gas price will be derived from the last element of the GasPriceChart with 2 decimal places precision.
-                       gasPriceStr := strconv.FormatFloat(skeleton.GasPriceChart.Data[len(skeleton.GasPriceChart.Data)-1]/1000000000, 'f', 2, 64)
+                       gasPriceStr := strconv.FormatFloat(meanGasPriceFirstBlock/1000000000, 'f', 2, 64)
                        skeleton.Current.Text = ui.GetCurrentText(skeleton.Current, ms.HeadBlock, gasPriceStr, ms.PeerCount, ms.ChainID, rpcUrl)
                }
 
@@ -587,7 +594,7 @@ func renderMonitorUI(ctx context.Context, ec *ethclient.Client, ms *monitorStatu
                }
 
                skeleton.TxPerBlockChart.Data = metrics.GetTxsPerBlock(renderedBlocks)
-               skeleton.GasPriceChart.Data = metrics.GetMeanGasPricePerBlock(renderedBlocks)
+               skeleton.GasPriceChart.Data = meanGasPricePerBlock
                skeleton.BlockSizeChart.Data = metrics.GetSizePerBlock(renderedBlocks)
                // skeleton.pendingTxChart.Data = metrics.GetUnclesPerBlock(renderedBlocks)
                skeleton.PendingTxChart.Data = observedPendingTxs.getValues(25)
```

Reply:
> I'm not sure if we want to take the last element of the array (the last block shown). It would maybe make more sense to use the first element instead (the first block shown), don't you think?
This was my initial take as well.. but it seemed a bit odd, because the graph's leftmost side would sometimes show `0` gas price, and the header gas price would show non-zero values. This made it look confusing, so I just stuck with the last element to show the current gas-price, instead of the historical gas-price.

Reply:
I think the first block of the selected range is shown on the right of the graph. The graph's leftmost side represents the older block in the selected range.

```bash
    Block t            Block t+1
       |                   |
-------------------------------->
```

Reply:
Also related to my first point I think you can just store the gas price per block in a variable and reuse it instead of computing it many times (which is more expensive) :)

Reply:
> I think the first block of the selected range is shown on the right of the graph. The graph's leftmost side represents the older block in the selected range.

I initially thought this would be the case as well.. but from the tests, it seems like the rightmost part of the graph represents the latest blocks, and this corresponds to the last element in the slice. Maybe I'm missing something here 

![image](https://github.com/user-attachments/assets/2042dd6e-54c9-44c5-9b8f-c1462fa39ee5)

Reply:
Yeah double-checked that and it's the case, lgtm! :)